core/src/main/java/com/sequenceiq/cloudbreak/init/clustertemplate/DefaultClusterTemplateCache.java

```diff
@@ -122,12 +132,69 @@ private void loadByClasspathPath(Collection<String> names) {
                 });
     }
 
-    private void convertToClusterTemplate(String templateAsString) throws IOException {
+    private void convertToClusterTemplate(String clusterTemplateName, String templateAsString,
+        Set<String> enabledPlatforms, Set<String> enabledGovPlatforms) throws IOException {
         DefaultClusterTemplateV4Request clusterTemplateRequest = new Json(templateAsString).get(DefaultClusterTemplateV4Request.class);
-        if (defaultClusterTemplates.get(clusterTemplateRequest.getName()) != null) {
-            LOGGER.warn("Default cluster template exists and it will be override: {}", clusterTemplateRequest.getName());
+        boolean useIt = doUseIt(clusterTemplateName, clusterTemplateRequest, enabledPlatforms, enabledGovPlatforms);
+        if (useIt) {
+            if (defaultClusterTemplates.get(clusterTemplateRequest.getName()) != null) {
+                LOGGER.warn("Default cluster template exists and it will be override: {}", clusterTemplateRequest.getName());
+            }
+            defaultClusterTemplates.put(
+                    clusterTemplateRequest.getName(),
+                    Base64.getEncoder().encodeToString(templateAsString.getBytes()));
+        }
+    }
+
+    private boolean doUseIt(String clusterTemplateName, DefaultClusterTemplateV4Request clusterTemplateRequest,
+        Set<String> enabledPlatforms, Set<String> enabledGovPlatforms) {
+        boolean useIt = true;
+        String aws = CloudPlatform.AWS.name();
+        boolean awsGovTemplate = clusterTemplateName.contains(GovUtils.GOV);
+        boolean awsGovEnabledOnDeployment = enabledGovPlatforms.contains(aws);
+        boolean awsCommercialEnabledOnDeployment = enabledPlatforms.contains(aws);
+        boolean localDeployment = awsGovEnabledOnDeployment && awsCommercialEnabledOnDeployment;
+        boolean govCloudDeployment = GovUtils.govCloudDeployment(enabledGovPlatforms, enabledPlatforms);
+        if (isAwsTemplate(clusterTemplateRequest, aws)) {
+            // the template version is AWS
+            if (isLocalDeploymentAndGovTemplate(awsGovTemplate, localDeployment)) {
+                // in case of local deployment we dont need to load gov templates
```

Comment:
why?

Reply:
because in that case aws templates are identical